### PR TITLE
[Distributed] Record generic substitution types for invocation

### PR DESCRIPTION
**Pull request by ktoso**
Record generic substitution types for generic functions in actors

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please build toolchain macOS

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please build toolchain macOS

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

**Comment by ktoso** [comment]
@swift-ci please build toolchain macOS

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

File: lib/SIL/Verifier/SILVerifier.cpp [hunk 1]
```diff
@@ -5528,6 +5528,11 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
           if (term->isFunctionExiting()) {
             require(state.Stack.empty(),
                     "return with stack allocs that haven't been deallocated");
+            if (!state.ActiveOps.empty()) {
+              for (auto op : state.ActiveOps) {
+                op->dump();
```
Comment (xedin) [inline lib/SIL/Verifier/SILVerifier.cpp:5533]: Should this be removed?

File: lib/SILGen/SILGenDistributed.cpp [hunk 2]
```diff
@@ -1015,20 +968,97 @@ void SILGenFunction::emitDistributedThunk(SILDeclRef thunk) {
...
+
+        // === Prepare generic signature
+        SubstitutionMap subs;
+        {
+          auto recordArgumentGenericSig =
+              recordGenSubFnDecl->getGenericSignature();
+          SmallVector<Type, 1> subTypes;
+          SmallVector<ProtocolConformanceRef, 2> subConformances;
+
+          //          subTypes.push_back(paramTy.getASTType());
```
Comment (xedin) [inline lib/SILGen/SILGenDistributed.cpp:1019]: Leftover commented code

File: lib/SILGen/SILGenDistributed.cpp [hunk 3]
```diff
@@ -1515,7 +1543,7 @@ void SILGenFunction::emitDistributedThunk(SILDeclRef thunk) {
           fnConv.getSILErrorType(getTypeExpansionContext()),
           OwnershipKind::Owned);
 
-      auto result = remoteCallReturnValue.getValue();
+      //      auto result = remoteCallReturnValue.getValue();
```
Comment (xedin) [inline lib/SILGen/SILGenDistributed.cpp:1546]: One more leftover commented code

**Review by xedin — COMMENTED** [review]
Looks reasonable to me, but I'm also no SIL expert :)

**Comment by ktoso** [comment]
Thanks! I'll cleanup the things you mentioned, missed a few it seems

**Comment by ktoso** [comment]
Thanks, fixing the cleanups right away in https://github.com/apple/swift/pull/41155/files

File: lib/SIL/Verifier/SILVerifier.cpp (hunk 1 above)
Comment (ktoso) [reply lib/SIL/Verifier/SILVerifier.cpp:5533]: yeah, though i'll admit the reality is adding it back every single time one is debugging sill issues like this... not in love with the pretty bad diagnostics that just "there was a bug" :p Might have another loop to do a pass or improving those diagnostics separately 🤔 

cleanup in https://github.com/apple/swift/pull/41155/

**Pull request merged**